esx_service/utils/vmdk_utils.py

```diff
@@ -274,26 +277,25 @@ def get_vm_uuid_by_name(vm_name):
     """ Returns vm_uuid for given vm_name, or None """
     si = vmdk_ops.get_si()
     try:
-        vm = [d for d in si.content.rootFolder.childEntity[0].vmFolder.childEntity if d.config.name == vm_name]
-        return vm[0].config.uuid
+        vm = FindChild(GetVmFolder(), vm_name)
+        return vm.config.uuid
     except:
         return None
 
 def get_vm_name_by_uuid(vm_uuid):
     """ Returns vm_name for given vm_uuid, or None """
     si = vmdk_ops.get_si()
     try:
-        vm = [d for d in si.content.rootFolder.childEntity[0].vmFolder.childEntity if d.config.uuid == vm_uuid]
-        return vm[0].config.name
+        return vmdk_ops.vm_uuid2name(vm_uuid)
```

Comment:
May be refactoring issue, but get VM UUID from name and get name from UUID should be in the vmdk_utils.py code. In this file we have the get UUID from name and the reverse is in vmdk_ops.py. Instead keep both variants here in vmdk_utils.py.

Not this issue but code is mixed.

From vmdk_ops,.py:
# Return VM managed object, reconnect if needed. Throws if fails twice.
def findVmByUuid(vm_uuid):
    si = get_si()
    vm = si.content.searchIndex.FindByUuid(None, vm_uuid, True, False)
    return vm

def vm_uuid2name(vm_uuid):
    vm = findVmByUuid(vm_uuid)
    if not vm or not vm.config:
        return None
    return vm.config.name